README.md

```diff
@@ -0,0 +1,34 @@
+# Grammar and parsing libraries for Rust
+
+This repository contains four Rust libraries for operating on Context-Free
+Grammars (CFGs) and parsing. Each can be used independently, or individually.
+The libraries are neither complete nor stable, but they may still be useful in
+their current form.
+
+  * `cfgrammar`: a library for loading in and manipulating CFGs. Currently only
+    supports grammars in Yacc format, though support for other formats may
+    follow in the future.
+
+  * `lrlex`: a basic lexer. It takes input files in a format similar, but
+    deliberately not identical, to `lex` and produces a simple lexer. Can be
+    used at compile-time (i.e. to produce `.rs` files, in similar fashion to
+    `lex`), or at run-time (i.e. dynamically loading grammars without producing
+    `.rs` files).
+
+  * `lrpar`: a parser for LR grammars. Takes Yacc files and produces a parser
+    that can produce a parse tree. Can be used at compile-time (i.e. to produce
+    `.rs` files, in similar fashion to `lex`), or at run-time (i.e. dynamically
+    loading grammars without producing `.rs` files).
+
+  * `lrtable`: a library for producing LR parse tables from grammars. Mostly
+    intended for use by `lrpar`, but may be useful separately for those who wish
```

Comment:
For use *with*?

Reply:
In retrospect, this whole description is a bit misleading. I've tried a bigger fix.